src/libraries/System.Text.Json/tests/Serialization/CustomConverterTests/CustomConverterTests.HandleNull.cs

```diff
@@ -0,0 +1,484 @@
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        [Fact]
+        public static void ValueTypeConverter_OptOut()
+        {
+            // Per null handling opt-out, serializer handles null.
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new Int32NullConverter_OptOut());
+
+            Assert.Equal(0, JsonSerializer.Deserialize<int>("null", options));
+        }
+
+        private class Int32NullConverter_OptOut : Int32NullConverter_SpecialCaseNull
+        {
+            public override bool HandleNull => false;
+        }
+
+        [Fact]
+        public static void ValueTypeConverter_NullOptIn()
+        {
+            // Per null handling opt-in, converter handles null.
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new Int32NullConverter_NullOptIn());
+
+            Assert.Equal(-1, JsonSerializer.Deserialize<int>("null", options));
+        }
+
+        private class Int32NullConverter_NullOptIn : Int32NullConverter_SpecialCaseNull
+        {
+            public override bool HandleNull => true;
+        }
+
+        [Fact]
+        public static void ComplexValueTypeConverter_NoOverride()
+        {
+            // Baseline
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<Point_2D_Struct>("null"));
+
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new PointStructConverter_SpecialCaseNull());
+
+            // Per null handling default value for value types (true), converter handles null.
+            var obj = JsonSerializer.Deserialize<Point_2D_Struct>("null", options);
+            Assert.Equal(-1, obj.X);
+            Assert.Equal(-1, obj.Y);
+        }
+
+        private class PointStructConverter_SpecialCaseNull : JsonConverter<Point_2D_Struct>
+        {
+            public override Point_2D_Struct Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                if (reader.TokenType == JsonTokenType.Null)
+                {
+                    return new Point_2D_Struct(-1, -1);
+                }
+
+                throw new JsonException();
+            }
+
+            public override void Write(Utf8JsonWriter writer, Point_2D_Struct value, JsonSerializerOptions options)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        [Fact]
+        public static void ComplexValueTypeConverter_OptOut()
+        {
+            // Per null handling opt-out, serializer handles null.
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new PointStructConverter_OptOut());
+
+            var obj = JsonSerializer.Deserialize<Point_2D_Struct>("null", options);
+            Assert.Equal(0, obj.X);
```

Comment:
As above, shouldn't this fail?

Reply:
https://github.com/dotnet/runtime/pull/35826/files#r421072296. 

Without the custom converter, [this will fail](https://github.com/dotnet/runtime/blob/b0329b19adf831cf81e9488c3de3547472784b2f/src/libraries/System.Text.Json/tests/Serialization/CustomConverterTests/CustomConverterTests.HandleNull.cs#L79).